### PR TITLE
Update documentation links

### PR DESCRIPTION
**Pull request by marcospereira**
1. Https all the things
2. Update links form Java 7 to Java 8
3. Update external documentation links
4. Correct markdown format for some links

### CODE REVIEW

File: documentation/manual/about/Philosophy.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!--- Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com> -->
 # Introducing Play 2
 
-Since 2007, we have been working on making Java web application development easier. Play started as an internal project at [Zenexity](http://www.zenexity.com) and was heavily influenced by our way of doing web projects: focusing on developer productivity, respecting web architecture, and using a fresh approach to packaging conventions from the start - breaking so-called JEE best practices where it made sense.
```
Comment (cchantep) [inline documentation/manual/about/Philosophy.md]: Name of the company was Zenexity by this time, so even if the website can be updated, the original name  should be used for me.

File: documentation/manual/detailedTopics/configuration/ws/CertificateGeneration.md [hunk 2]
```diff
@@ -10,9 +10,9 @@ The best way to think about public key certificates is as a passport system. Cer
 
 keytool comes in several versions:
 
-* [1.8](http://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html)
-* [1.7](http://docs.oracle.com/javase/7/docs/technotes/tools/solaris/keytool.html)
-* [1.6](http://docs.oracle.com/javase/6/docs/technotes/tools/solaris/keytool.html)
+* [1.8](https://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html)
```
Comment (cchantep) [inline documentation/manual/detailedTopics/configuration/ws/CertificateGeneration.md]: As the support of Java 7 is ended, I would guess keytool 1.8 should be the minimal corresponding version.

File: documentation/manual/working/javaGuide/advanced/dependencyinjection/JavaDependencyInjection.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ Out of the box, Play provides dependency injection support based on [JSR 330](ht
 
 ## Declaring dependencies
 
-If you have a component, such as a controller, and it requires some other components as dependencies, then this can be declared using the [@Inject](http://docs.oracle.com/javaee/6/api/javax/inject/Inject.html) annotation.  The `@Inject` annotation can be used on fields or on constructors, which you decide to use is up to you.  For example, to use field injection:
+If you have a component, such as a controller, and it requires some other components as dependencies, then this can be declared using the [@Inject](https://docs.oracle.com/javaee/7/api/javax/inject/Inject.html) annotation.  The `@Inject` annotation can be used on fields or on constructors, which you decide to use is up to you.  For example, to use field injection:
```
Comment (cchantep) [inline documentation/manual/working/javaGuide/advanced/dependencyinjection/JavaDependencyInjection.md]: 8 ?

File: documentation/manual/working/scalaGuide/advanced/http/HttpApi.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ Basically, Play2 is an API that abstractly have the following type:
 RequestHeader -> Array[Byte] -> Result 
 ```
 
-The above [computation](http://www.haskell.org/arrows/) takes the request header `RequestHeader`, then takes the request body as `Array[Byte]` and produces a `Result`.
```
Comment (cchantep) [inline documentation/manual/working/scalaGuide/advanced/http/HttpApi.md]: Is this (repeated) link to Haskell arrow useful to understand Iteratee?

File: documentation/manual/about/Philosophy.md (hunk 1 above)
Comment (marcospereira) [inline documentation/manual/about/Philosophy.md]: We can use both here. Perhaps something like "now known as zengularity".

Comment (cchantep) [inline documentation/manual/about/Philosophy.md]: "Zenexity (now [Zengularity](http://www.zengularity.com))" is sufficient I think

Comment (marcospereira) [inline documentation/manual/about/Philosophy.md]: Agree.

File: documentation/manual/working/javaGuide/advanced/dependencyinjection/JavaDependencyInjection.md (hunk 3 above)
Comment (marcospereira) [inline documentation/manual/working/javaGuide/advanced/dependencyinjection/JavaDependencyInjection.md]: This is JEE. Version 8 wasn't released yet, was it?

File: documentation/manual/detailedTopics/configuration/ws/CertificateGeneration.md (hunk 2 above)
Comment (marcospereira) [inline documentation/manual/detailedTopics/configuration/ws/CertificateGeneration.md]: Good point, but I'm not sure if old certificates will work properly with the new java version. If they won't, we need to explain it at the migration guide.

File: documentation/manual/about/Philosophy.md (hunk 1 above)
Comment (marcospereira) [inline documentation/manual/about/Philosophy.md]: Fixed.

File: documentation/manual/detailedTopics/configuration/ws/CertificateGeneration.md (hunk 2 above)
Comment (marcospereira) [inline documentation/manual/detailedTopics/configuration/ws/CertificateGeneration.md]: Fixed.

**Comment by jroper** [comment]
By the way, did you you can use `sbt validateExternalLinks`, and it will go and make sure none of the external links are dead?  We don't run it in CI because it would make our build too unstable, and also some sites (in particular, meetup.com) think the link checker is a bot trying to crawl their site and start blocking the requests.  But it's a convenient way to find dead links in the documentation.

**Comment by marcospereira** [comment]
@jroper did not know that and wrote my own script. :-)

**Pull request merged**